client/app/reducers/selection.coffee

```diff
@@ -0,0 +1,38 @@
+Immutable = require 'immutable'
+
+{ActionTypes} = require '../constants/app_constants'
+
+DEFAULTSTATE =
+    selected: Immutable.Set()
+    allSelected: false
+
+module.exports = (state = DEFAULTSTATE, action) ->
+    switch action.type
+        when ActionTypes.MAILBOX_SELECT_ALL
+            if state.allSelected
+                nextState = DEFAULTSTATE
```

Comment:
Until we move on to saga or another redux middleware, actions should be considered as events, ideally in the past tense ("SOMETHING_HAPPENED"). This action "full" name could be `USER_CLICKED_SELECT_ALL_BUTTON`, if we add more button, we can add more actions. 
if another situation require us to select all messages (for instance, going to next page with select all enable across pages), we should handle it in the appropriate action ("USER_CLICKED_NEXT_PAGE")

The other extreme is to have actions as setters (SET_CURRENT_MESSAGE), in which case, you end up calling several action in a row from a ActionCreator, you have to be sure you call them in the proper order at call site, you have to be sure you are calling all of them and so on.